src/AuthoringUtils.ts

```diff
@@ -32,107 +55,139 @@ export class AuthoringUtils {
+            docFragment.appendChild(linkElement);
+        });
+
+        return docFragment;
+    }
+
+    private generateScriptElements(jsUrls: string[]): DocumentFragment {
+        const docFragment: DocumentFragment = document.createDocumentFragment();
+
+        jsUrls.forEach((url: string) => {
+            const htmlScriptElement = document.createElement('script');
+
+            htmlScriptElement.type = 'text/javascript';
+            htmlScriptElement.src = url;
+            docFragment.appendChild(htmlScriptElement);
+        });
+
+        return docFragment;
     }
 
     /**
-     * Returns string value of all the concatenated tags.
-     * @returns Concatenated tags.
+     * Checks if edit mode is on.
+     * @returns `true` if application is in AEM `EDIT` mode.
      */
-    private generateElementString(tagType: string, attr: string, attrValue: string): string {
-        let tag = '';
-
-        if (tagType === TAG_TYPE.JS) {
-            tag = `<script ${attr}="${attrValue}"></script>`;
-        } else if (tagType === TAG_TYPE.STYLESHEET) {
-            tag = `<link ${attr}="${attrValue}"/>`;
-        }
+    public static isEditMode(): boolean {
+        const viaMetaProperty = PathUtils.getMetaPropertyValue(MetaProperty.WCM_MODE) === AEM_MODE.EDIT;
+        const viaQueryParam = PathUtils.isBrowser() && (AuthoringUtils.getWCMModeFromURL() === AEM_MODE.EDIT);
 
-        return tag;
+        return viaMetaProperty || viaQueryParam;
     }
 
     /**
-     * Checks status of requested state.
-     * @returns `true` if application is in authoring state and AEM mode is `EDIT`.
+     * Checks if app is a remote application.
+     * If the cq:wcmmode is provided as get parameter it is implied that the app is remote.
+     * @returns `true` if the application is a remote app.
      */
-    public static isStateActive(state: string): boolean {
-        if (state === Constants.STATE_AUTHORING) {
-            const viaMetaProperty = PathUtils.getMetaPropertyValue(MetaProperty.WCM_MODE) === AEM_MODE.EDIT;
-            const viaQueryParam = PathUtils.isBrowser() && (AuthoringUtils.getAemMode() === AEM_MODE.EDIT);
+    public static isRemoteApp(): boolean {
+        try {
+            const url = new URL(PathUtils.getCurrentURL());
 
-            return viaMetaProperty || viaQueryParam;
+            return !!url.searchParams.get(MetaProperty.WCM_MODE);
+        } catch (e) {
+            // invalid url
         }
 
         return false;
     }
 
     /**
-     * Checks AEM mode.
+     * Checks AEM mode from URL.
      * @private
-     * @returns AEM mode or `null`.
+     * @returns AEM mode.
      */
-    public static getAemMode(): string | null {
+    private static getWCMModeFromURL(): string {
         let url: URL;
 
         try {
             url = new URL(PathUtils.getCurrentURL());
 
-            return url.searchParams.get(Constants.AEM_MODE_KEY);
+            return url.searchParams.get(MetaProperty.WCM_MODE) || '';
```

Comment:
The name of the constant is misleading as it suggests to be used with a meta property. What do you suggest?

Reply:
I thought it's acceptable this way, but I am fine changing it. 
The reasoning behind acceptable is:
- Function name is getWCMModeFromURL
- javadoc says it's from url

Alternative solutions:
- Rename MetaProperty to more generic name contants class (not binding to Meta)
- Create a new constant class
- Add more javadoc 

I am fine with all, please let me know what you like best.